lib/thetvdb.py

```diff
@@ -437,17 +437,18 @@ def get_kodi_unaired_episodes(self, single_episode_per_show=True, include_last_e
         kodi_series = self.get_kodishows(True)
         next_episodes = []
         for kodi_serie in kodi_series:
-            serieid = kodi_serie["tvdb_id"]
-            if single_episode_per_show:
-                episodes = [self.get_nextaired_episode(serieid)]
-            else:
-                episodes = self.get_unaired_episodes(serieid)
-            if include_last_episode:
-                episodes.append(self.get_last_episode_for_series(serieid))
-            for next_episode in episodes:
-                if next_episode:
-                    # make the json output kodi compatible
-                    next_episodes.append(self._map_kodi_episode_data(kodi_serie, next_episode))
+            if kodi_serie["tvshowid"] in tvshows_ids:
```

Comment:
In relation to the previous comment, this line should be:

if not tvshow_ids or (tvshow_ids and kodi_serie["tvshowid"] in tvshows_ids):